sw/device/lib/base/mmio.c

```diff
@@ -4,6 +4,122 @@
 
 #include "sw/device/lib/base/mmio.h"
 
+#include <stdalign.h>
+
+#include "sw/device/lib/base/memory.h"
+
+/**
+ * Computes how many bytes `addr` is ahead of the previous 32-bit word alignment
+ * boundary.
+ */
+static ptrdiff_t misalignment32_of(uintptr_t addr) {
+  return addr % alignof(uint32_t);
```

Comment:
I wonder if the compiler optimizes this to `addr & (sizeof(uint32_t) - 1)`.

Reply:
Wonder no more! https://godbolt.org/z/dWKKBS (Short answer: the code used to look like your comment, but it's clearer how it's written here and we knew this optimisation would be performed).

Reply:
Ohh, nice! thanks.